src/main/java/recipeapplication/database/RecipeFirebaseCrud.java

```diff
@@ -61,6 +65,36 @@ public Recipe delete(String document) {
         return deletedRecipe;
     }
 
+    public Recipe[] queryCategory(String category) {
+        CollectionReference recipes = firestore.collection(collection);
+        Query query = recipes.whereEqualTo("category", category.toUpperCase());
+
+        return getRecipeArrayFromQuery(query).toArray(new Recipe[0]);
+
+    }
+
+    public Recipe[] queryIngredients(String[] ingredients) {
+        CollectionReference recipes = firestore.collection(collection);
+        Query query = recipes.whereArrayContainsAny("ingredients", Arrays.asList(ingredients));
+
+        ArrayList<Recipe> recipeArrayBeforeFilter = getRecipeArrayFromQuery(query);
+        ArrayList<Recipe> recipeArrayAfterFilter = new ArrayList<>();
+        for (Recipe recipe : recipeArrayBeforeFilter) {
+            if(StringUtils.isFullyContainedInArray(recipe.getIngredients(), Arrays.asList(ingredients))) {
+                recipeArrayAfterFilter.add(recipe);
+            }
+        }
+        return recipeArrayAfterFilter.toArray(new Recipe[0]);
```

Comment:
Related to the previous comment, make a method that accepts both Ingredients and category, check if they are not empty, and filter the request by the non-empty parameters

Reply:
Issue has been fixed